cmd/thriftrw-list-deps/main.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/jessevdk/go-flags"
+
+	"go.uber.org/thriftrw/compile"
+)
+
+type args struct {
+	ThriftFile string `positional-arg-name:"thrift-file"`
+}
+
+var opts struct {
+	Relative string `long:"relative" description:"Output paths will be relative to this directory"`
+	Args     args   `positional-args:"yes" required:"yes"`
+}
+
+func do(relative string, inputThriftFile string, output io.Writer) error {
+	module, err := compile.Compile(inputThriftFile)
+	if err != nil {
+		return err
+	}
+
+	for _, included := range module.Includes {
+		outputFilePath := included.Module.ThriftPath
+		if relative != "" {
+			outputFilePath, err = filepath.Rel(relative, outputFilePath)
+			if err != nil {
+				return err
+			}
+		}
+		fmt.Fprintf(output, "%s\n", outputFilePath)
+	}
+
+	return nil
+}
+
+func main() {
+	if _, err := flags.Parse(&opts); err != nil {
+		log.Fatalf("error parsing arguments: %s", err.Error())
+		os.Exit(1)
+	}
+
+	inputThriftFile := opts.Args.ThriftFile
+
+	if err := do(opts.Relative, inputThriftFile, os.Stdout); err != nil {
```

Comment:
We should try to have exactly one log.Fatal in the program. The way I like to organize it is,

```go
func main() {
  log.SetFlags(0)  // so that the error message isn't noisy
  if err := run(); err != nil {
    log.Fatalf("%+v", err)  // this exits with a non-zero code
  }
}

// run is the real "main", except it returns an error instead of calling log.Fatal
func run() error {
  var opts struct {
    // flags defined here
  }
  // parse flags, return err if non-nil.

  deps, err := listDependencies(opts.RelativeTo, opts.InputFile)
  if err != nil {
    ...
  }

  for _, dep := range deps {
    fmt.Println(dep)
  }
}

func listDependencies(...) ([]string, error) {
   // business logic
}
```

Then tests can be written against `listDependencies`.